core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -2424,8 +2424,12 @@ private void scheduleAsyncPersistenceInternal(LockedInodePath inodePath) throws
   public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFiles)
       throws FileDoesNotExistException, InvalidPathException, AccessControlException {
     for (long fileId : persistedFiles) {
-      // Permission checking for each file is performed inside setAttribute
-      setAttribute(getPath(fileId), SetAttributeOptions.defaults().setPersisted(true));
+      try {
+        // Permission checking for each file is performed inside setAttribute
+        setAttribute(getPath(fileId), SetAttributeOptions.defaults().setPersisted(true));
+      } catch (FileDoesNotExistException | AccessControlException | InvalidPathException e) {
+        LOG.error("Failed to set file {} as persisted, because {}", fileId, e);
+      }
```

Comment:
Ideally, yes. However, that would require bigger architectural change. The file could be deleted/renamed or under other metadata change, and we want the metadata change to be propagated to ufs asynchronously.